### DISCUSSION TITLE
ARM 2.6.71 working under Ubuntu 23.10 Mantic Minotaur

### DISCUSSION THREAD

**Discussion by Greybeard-1963** [automatic-ripping-machine/automatic-ripping-machine#1072] [General]
I have successfully got ARM 2.6.71 working under Mantic Minotaur, this workaround may also be applicable to other distros and versions of ARM. 

If you follow the the Wiki instructions, the container will install successfully etc. but the web interface will fail to come up (normally on localhost:8080 or HOST_IP:8080). Leaving the container running will eventually cause it to report ARM to be "Unhealthy".

The reason for this is that the directory tree is not being allotted the correct permissions, these [IIRC] were root:root rather than arm:arm. Changing these permissions fixes this problem and allows the container to initialise properly.  

To run ARM under the latest Ubuntu:

[1] Follow the  [Wiki instructions](https://github.com/automatic-ripping-machine/automatic-ripping-machine/wiki/docker) install a docker image, containers etc, and run `start_arm_container.sh` as instructed from within the directory /home/arm.

At this point, ARM will start and you will see the container running:

[2] `sudo docker container list`  

Take a note of the relevant 12 digit CONTAINER_ID. You will now need to stop the container:

[3]  `sudo docker container stop CONTAINER_ID` (CONTAINER_ID being the 12 digit hash from step [2]).

Now assuming you have specified all the necessary directories for ARM are kept under the sub-directory /home/arm/ARM (e.g. in  start_arm_container.sh there is the line  `-v "/home/arm/ARM:/home/arm" \` ), execute the following:

[4] `sudo chown arm:arm /home/arm/ARM -vR`

This will produce a list of files with arm:arm ownership and the files/dirs that have been updated.

Now restart the container using the ID from step [1]:

[5]  `sudo docker container start CONTAINER_ID`

Check that the container is running:

[6] `sudo docker container list`  

Check that the webserver is running (you will need nmap installed for this):

[7] `nmap localhost`

You should see port 8080 open (or whatever port you have requested ARM to reside on).

8080/tcp open  http-proxy

I haven't had time to test all the ARM functionality yet, but ARM managed to rip an audio CD successfully. 

YMMV.

**Comment by microtechno9000** [comment]
Glad to see you got it working

I updated the Wiki yesterday to mention the ownership issue and how to resolve. Sorry didn't post about that in this issue.

Hopefully it helped. If there are missing steps or bits of information feel free to amend the wiki, instructions are here [ARM Wiki Guide](https://github.com/automatic-ripping-machine/automatic-ripping-machine/wiki/Contribute-Wiki)

**Reply by Greybeard-1963** [reply]
Thank you, much appreciated. I'm not sure if this is a docker container or ARM issue, so I haven't flagged it as a bug but it is easily fixed at the CLI. 

Next step I'll test it on 22.04 server, to try and maximize space for all the rips. Naturally I'll report back if I find anything important.

**Comment by Tippon** [comment]
@Greybeard-1963 Thank you for this. I've been trying to get ARM working on Xubuntu 22.04.4 LTS and was getting the permissions issues. I've just used your steps and ARM is working now.

I don't know if I've messed something up when I've been trying to get it working, but my arm.yaml is under /home/arm/config/ and the IP address and port didn't match my settings. It's possible that I wiped those by reinstalling, but I'm pretty sure that I set them again. The IP address was set to my external IP rather than localhost or 127.0.0.1, or my fixed internal IP.

I'm still adjusting the settings before I test it with a disc, but I thought I'd let you and @microtechno9000 know about this in case it affects anything and isn't just me getting things wrong 😁

**Reply by microtechno9000** [reply]
If running in docker, the IP config gets ignored as docker just exposes the port. However, the arm.yml config isn't removed when you update that remains constant as a docker volume

**Reply by Tippon** [reply]
Ah, ok. Thank you, I'm still learning about Docker :)

**Reply by microtechno9000** [reply]
No worries
It is a lot to learn. At the end of this if you have any suggestions of useful information for the Wiki please post it here. Even if its just a dot point list, its hard to know after working on the project what information would be useful to new players.

**Comment by Greybeard-1963** [comment]
No problem, @Tippon glad to be of help. I can also confirm that ARM runs perfectly (Ripping audio & dvd's) under Ubuntu server 22.04.4 LTS if installed with the above workaround.
